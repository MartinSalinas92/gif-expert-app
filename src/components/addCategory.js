
import React, {useState} from 'react'
import PropTypes from 'prop-types';

 const AddCategory = ({setCategories}) => {

    const [state, setstate] = useState('');

    const cambioTexto= (e)=>{

        //console.log(e.target.value);

        setstate(e.target.value)


    }

    const pushEnter= (e)=>{

        e.preventDefault();

        console.log(pushEnter);

        if(state.trim().length > 2){

            setCategories(cats => [state, ...cats])

        }

      

       
    }


    return (
        <form onSubmit={pushEnter}>
            <p>{state}</p>
                <input
                type="text" 
                value={state}
                onChange={cambioTexto}
                
                
                />
            
        </form>
    )
}

AddCategory.propTypes={

    setCategories: PropTypes.func.isRequired

}
export default AddCategory


