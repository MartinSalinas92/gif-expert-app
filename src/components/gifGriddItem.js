import React from 'react';
import PropTypes from 'prop-types';

export function GifGriddItem({img, title, images}) {

   
    return (
        <div className="card animate__animated animate__bounce">
             <img src={images} alt={title} />

             <p className="animate__animated animate__fadeInLeft">{title}</p>
            
            
        </div>
    )
}

GifGriddItem.propTypes={

    title:PropTypes.string.isRequired,
    images:PropTypes.string.isRequired
}





