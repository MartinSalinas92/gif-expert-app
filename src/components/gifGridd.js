import React from 'react';
import { useFetchGifs } from '../Hooks/useFetchGIf';
import { GifGriddItem } from './gifGriddItem';
import PropTypes from 'prop-types';

export const GifGridd = ({category}) => {

    

   const {data, loading}= useFetchGifs(category);

   
  

  

  

   

    return (
        <>
            <h3>{category}</h3>



           
                
            <div className="card-grid">

                {loading && <p>loading</p>}

            {
                data.map(img=>(

                    <GifGriddItem
                      key={img.id}  

                      {...img}

                    
                    />
                    
                ))
            }
                  
            
                
                    
            </div>
                    
                
           
           
        </>
    )
}

GifGridd.propTypes={

    category:PropTypes.string.isRequired

}
