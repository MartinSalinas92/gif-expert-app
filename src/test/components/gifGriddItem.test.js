import { shallow } from "enzyme"
import { GifGriddItem } from "../../components/gifGriddItem"

describe('Pruebas componente <gifGriddItem />', ()=>{

    const title="martin";
    const images="http://holajulio.com";
    
    const wrapper= shallow(<GifGriddItem title={title} images={images} />)

    test('Debe de mostrar el componente nuevamente ', () => {

        expect(wrapper).toMatchSnapshot()
        
    });
    test('Debe de tener un parrafo con el title', () => {

       const p= wrapper.find('p');
        expect(p.text().trim()).toBe(title);
        
    });
    test('debe tener la imagen igual al url y alt de los props', () => {

        const img= wrapper.find('img');

        //console.log(img.prop('src'));

        expect(img.prop('src')).toBe(images);
        expect(img.prop('alt')).toBe(title);
        
    });
    test('debe de tener animate__bounce', () => {

        const div=wrapper.find('div');

        //console.log()

        const className=div.prop('className');

        expect(className.includes('animate__bounce')).toBe(true);


        
    });
    
    
    
    
})