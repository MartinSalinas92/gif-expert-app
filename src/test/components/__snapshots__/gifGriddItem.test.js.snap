// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente <gifGriddItem /> Debe de mostrar el componente nuevamente  1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="martin"
    src="http://holajulio.com"
  />
  <p
    className="animate__animated animate__fadeInLeft"
  >
    martin
  </p>
</div>
`;
